docs/design/dynamic-gpu-slice.md

```diff
@@ -0,0 +1,122 @@
+# NVIDIA GPU MPS and MIG dynamic slice plugin
+
+## Introduction
+
+The NVIDIA GPU build-in sharing method includes: time-slice, MPS and MIG. The context switch for time slice sharing would waste some time, so we chose the MPS and MIG. The GPU MIG profile is variable, the user could acquire the MIG device in the profile definition, but current implementation only defines the dedicated profile before the user requirement. That limits the usage of MIG. We want to develop an automatic slice plugin and create the slice when the user require it. And we also add the MPS support.
+For the scheduling method, node-level binpack and spread will be supported. Referring to the binpack plugin, we consider the CPU, Mem, GPU memory and other user-defined resource.
+
+## Targets
+
+- CPU, Mem, and GPU combined schedule
+- GPU dynamic slice: includes MPS and MIG
+- Support node-level binpack and spread by GPU memory
+
+## Limitations
+
+- X86 structure
+- Kubernetes 1.26 or higher
+
+## Structure
+
+![](images/dynamicGPUSliceStructure.png)
+
+### Config maps
+- volcano-scheduler-configmap
+This configmap defines the plugin configurations including policy, weight, and MIG geometries.
+
+```yaml
+apiVersion: v1
+data:
+  volcano-scheduler.conf: |
+    actions: "enqueue, allocate, backfill"
+    tiers:
+    - plugins:
+      - name: cugpushare
+        arguments:
+          cugpushare.schedulePolicy: spread
+          cugpushare.scheduleWeight: 10
+          cugpushare.weight.cpu: 1
+          cugpushare.weight.memory: 2
+          cugpushare.weight.gpu: 5
+          cugpushare.DevicePluginCMName: mps-configmap
```

Comment:
Why do we need to specify the name and ns of the configmap here, isn't this fixed?

Reply:
This is a shared config between the planner and the device plugin. The name could be configured. The planner(in the scheduler) and device plugin should define the same name configmap.